docs/pages/access-controls/device-trust/enforcing-device-trust.mdx

```diff
@@ -148,6 +152,56 @@ cluster. Likewise, a root cluster will not enforce device trust on resources in
 leaf clusters.
 </Admonition>
 
+## App Access support
+
+Apps may enforce device trust via [role-based enforcement](
+#role-based-trusted-device-enforcement). Any apps that enforce device trust must
+be accessed using tsh commands such as [tsh proxy app](
+../../reference/cli/tsh.mdx#tsh-proxy-app) or using the certificates issued by
+`tsh app login`. Direct access to those apps via the Web UI is not yet
+available.
+
+For example, to enforce device trust for all `env:production` apps, save the
+role below as `require-trusted-device-apps.yaml`:
+
+```yaml
+kind: role
+version: v7
+metadata:
+  name: require-trusted-device-apps
+spec:
+  options:
```

Comment:
How did you end up doing this? Most of our role options apply unconditionally to anyone who has the role, but it looks like in this case the `device_trust_mode` option only applies when there's a match on `app_labels`?

Reply:
I stole it from require_session_mfa, which also only applies to the labels. :)

Reply:
You can follow the trail from here, if you are curious: https://github.com/gravitational/teleport/blob/cc7608b9da600a7f9da4fc5ce2113b6aafd44653/lib/services/role.go#L2426.

Reply:
The device trust role option always works like that, it's mentioned in the role-based section: https://github.com/gravitational/teleport/pull/35335/files#diff-50ffd70881a0efbcd6a76407915f5e65f2b4bf746e3c953991e453b8b2bbdb9fR39-R42

I'll take a look at the roles page, keeping this open to remind me.